site/package.json

```diff
@@ -52,34 +52,36 @@
     "@babel/preset-env": "^7.24.3",
     "@babel/preset-typescript": "^7.22.11",
     "@cypress/skip-test": "^2.6.1",
+    "@stylistic/eslint-plugin": "^2.1.0",
     "@types/jest": "^29.5.4",
     "@types/jquery": "^3.5.29",
     "@types/jquery.are-you-sure": "^0.0.34",
     "@types/jqueryui": "^1.12.22",
+    "@types/node": "^20.14.2",
     "@types/twig": "^1.12.16",
-    "@typescript-eslint/eslint-plugin": "^7.7.1",
-    "@typescript-eslint/parser": "^7.11.0",
     "babel-jest": "^29.7.0",
     "cypress": "^13.6.4",
     "cypress-file-upload": "^5.0.8",
     "esbuild": "^0.21.1",
     "eslint": "^8.57.0",
-    "eslint-plugin-cypress": "^2.15.1",
-    "eslint-plugin-jest": "^28.3.0",
+    "eslint-plugin-cypress": "^3.3.0",
+    "eslint-plugin-jest": "^28.6.0",
+    "globals": "^15.4.0",
     "jest": "^29.6.4",
     "jest-environment-jsdom": "^29.7.0",
     "node-fetch": "^2.6.12",
     "regenerator-runtime": "^0.14.0",
     "stylelint": "^15.11.0",
     "stylelint-config-standard": "^33.0.0",
     "typescript": "^5.4.5",
+    "typescript-eslint": "^7.13.0",
     "vnu-jar": "^23.4.11"
   },
   "scripts": {
     "css-stylelint": "stylelint \"**/*.css\"",
     "css-stylelint:fix": "stylelint \"**/*.css\" --fix",
-    "eslint": "eslint \"public/js/**/*.js\" \"cypress/**/*.js\" \"ts/**/*.js\" \"ts/**/*.ts\" \"tests/ts/**/*.js\"",
```

Comment:
Will this now try and lint the node_modules folder? (or other folders that aren't ours to fix)

Reply:
It hasn't on my machine or the CI, so no. The only files that would be linted are the ones specified in `eslint.config.js`, and eslint seems to auto-exclude node_modules. Ill edit eslint.config.js to be more explicit though.

Reply:
Ok with dba82dcfa0ea1a8524e9acccb3e21a7d374856f7 I made it so that eslint lints all js/ts files by default and ignores ones we aren't supposed to lint. The config block with `ignores` by itself is applies to the entire config, so even with running eslint with `eslint .` it will still ignore things it's supposed to.